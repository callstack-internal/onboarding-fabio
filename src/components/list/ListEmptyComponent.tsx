import React from 'react';
import {Pressable, StyleSheet, Text} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
import Colors from '../../styles/Colors';

type Props = {
  description: string;
  onTryAgainPress: () => void;
};

const ListEmptyComponent = ({description, onTryAgainPress}: Props) => {
  return (
    <SafeAreaView style={styles.rootContainer} edges={['left', 'right']}>
      <Text style={styles.descriptionLabel}>{description}</Text>

      <Pressable
        style={styles.tryAgainButtonContainer}
        accessibilityRole={'button'}
        onPress={onTryAgainPress}>
        <Text style={styles.tryAgainButtonLabel}>Try Again</Text>
        <MaterialIcon name={'refresh'} size={24} color={Colors.blue} />
      </Pressable>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  rootContainer: {
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 40,
  },

  descriptionLabel: {
    fontSize: 14,
    color: Colors.black,
  },

  tryAgainButtonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 20,
  },
  tryAgainButtonLabel: {
    fontSize: 14,
    fontWeight: 'bold',
    color: Colors.blue,
  },
});

export default ListEmptyComponent;
