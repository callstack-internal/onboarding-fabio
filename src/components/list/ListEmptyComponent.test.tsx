import React from 'react';
import ListEmptyComponent from './ListEmptyComponent';

test('should render a different state correctly', () => {
  expect(
    <ListEmptyComponent
      description={'Empty list.'}
      onTryAgainPress={() => {}}
    />,
  ).toMatchDiffSnapshot(
    <ListEmptyComponent
      description={'Another description.'}
      onTryAgainPress={() => {}}
    />,
  );
});
