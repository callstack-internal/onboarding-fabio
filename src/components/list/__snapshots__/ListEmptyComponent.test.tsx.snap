// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a different state correctly 1`] = `
"Snapshot Diff:
- <ListEmptyComponent description="Empty list." onTryAgainPress={[Function onTryAgainPress]} />
+ <ListEmptyComponent description="Another description." onTryAgainPress={[Function onTryAgainPress]} />

@@ -19,11 +19,11 @@
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
-     Empty list.
+     Another description.
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {"
`;
